staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -74,6 +74,10 @@ type DiscoveryInterface interface {
 	ServerVersionInterface
 	OpenAPISchemaInterface
 	OpenAPIV3SchemaInterface
+	// Returns copy of current discovery client that will only
+	// receive the legacy discovery format. Could return nil
```

Comment:
I would prefer it if it just returned itself instead of nil if it doesn't implement this? Hard to chain calls to things that might return nil.